include/aspect/utilities.h

```diff
@@ -1079,17 +1078,30 @@ namespace aspect
     {
       public:
         /**
-         * Constructor based on providing the geometry model as a pointer
+         * Constructor based on providing the geometry model as a pointer.
          */
         NaturalCoordinate(Point<dim> &position,
                           const GeometryModel::Interface<dim> &geometry_model);
 
+        /**
+         * Constructor based on providing the coordinates and associated
+         * coordinate system.
+         */
+        NaturalCoordinate(std_cxx11::array<double, dim> &coord,
+                          const Utilities::Coordinates::CoordinateSystem &coord_system);
+
         /**
          * Returns the coordinates in the given coordinate system, which may
          * not be Cartesian.
          */
         std_cxx11::array<double,dim> &get_coordinates();
 
+        /**
+         * Returns the coordinates in the given coordinate system, which may
+         * not be Cartesian. Output is a Point.
+         */
+        Point<dim> get_point();
```

Comment:
this function is needed because deal.II parsedfunction evaluate() expects a Point

Reply:
I am not sure if this is necessary, since you can also use the utilities function `convert_array_to_point` to do this.